_posts/2017-02-09-whats-bonnyci-wheres-it-going.markdown

```diff
@@ -0,0 +1,170 @@
+---
+layout: page
+title:  "What is BonnyCI? Where's it going?"
+categories:
+  - bonnyci
+tags:
+  - zuul
+  - ci
+  - testing
+  - gating
+---
+
+Avast! Welcome, and thanks for your interest in BonnyCI. It's about time
+we publicly explain what BonnyCI is, and why it works the way it does.
+
+
+The story begins with [OpenStack](http://www.openstack.org/). Early
+on in OpenStack's history, some of the
+developers identified a need for robust [Continuous
+Integration](https://en.wikipedia.org/wiki/Continuous_integration).
+This included testing multiple projects together to ensure
+they interact well. Developers who set this system up chose
+[Jenkins](http://jenkins.io/), and also decided to pursue a path of gated
+pre-merge CI, requiring every commit to pass the defined integration
+tests before landing.
+
+This proved to be a good decision, as it kept all source repositories
+more or less usable by the development community, which exploded in size
+rapidly. However, this presented a problem.  Gated CI meant that every
+commit from every project would need to run the entire test suite.
+
+This has some trade-offs that OpenStack's development community
+did not want to accept. Landing patches would be inherently slow,
+and it would discourage adding test coverage. Even running the
+matrix of different tests in parallel would mean that commits
+can land no faster than the longest and most comprehensive test
+takes. Why bother with each commit? In addition to the benefits for
+developers to be on the same page and able to confidently start
+from a working place, many OpenStack users embrace [Continuous
+Delivery](https://en.wikipedia.org/wiki/Continuous_delivery). This
```

Comment:
If we are going to link this more than once we should consider using [reference links][1]. [Reference links][1] let us easily link to a resource repeatedly.

While we are using [reference links][1] for this, maybe we should for other linked resources too?

[1]: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links